python/ray/serve/api.py

```diff
@@ -56,7 +56,8 @@ def __call__(self, *, python_arg=None):
     return f
 
 
-def init(name: Optional[str] = None,
+# TODO(architkulkarni): Add type hint for name after upgrading cloudpickle.
```

Comment:
this is fine. I think cloudpickle only breaks for `Optional` in generated class like `RayServeWrappedWorker`

Reply:
I see, that makes sense.  I forgot that the tests for the type hints for API were passing earlier.  I'll revert this change